test/fitnesse/junit/JUnitXMLPerPageTestListenerIntegrationTest.java

```diff
@@ -0,0 +1,213 @@
+package fitnesse.junit;
+
+import org.gradle.api.tasks.testing.TestDescriptor;
+import org.gradle.api.tasks.testing.TestResult;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class JUnitXMLPerPageTestListenerIntegrationTest {
```

Comment:
Given you have tested the xml writer separately I would suggest we keep only the the IntegrationTest or test with mocks.

Reply:
Deleted integration test, kept unit test.